### PR TITLE
Fix issue with MACRO support

### PR DESCRIPTION
**Pull request by TurboGit**
Also add some INLINE before/after separator in output.

### CODE REVIEW

File: src/templates_parser-print_tree.adb [hunk 1]
```diff
@@ -78,6 +81,30 @@ procedure Print_Tree (T : Tree; Level : Natural := 0) is
...
    end Print_Indent;
 
+   ---------------
+   -- Sep_Image --
+   ---------------
+
+   function Sep_Image (Str : Unbounded_String) return String is
+      R : Unbounded_String;
+   begin
+      for K in 1 .. Length (Str) loop
```
Comment (leogermond) [inline src/templates_parser-print_tree.adb:91]: Replacing the 3 lines by
```
for E of To_String (Str) loop
```
would make the code clearer, and potentially more efficient (due to skipping indexing altogether)

File: src/templates_parser-print_tree.adb [hunk 2]
```diff
@@ -227,8 +254,13 @@ begin
          Print_Tree (T.Next, Level);
 
       when Inline_Stmt =>
```
Comment (leogermond) [inline src/templates_parser-print_tree.adb:256]: Guaranteed to always have a `Before` and `After` nodes?

File: src/templates_parser.adb [hunk 3]
```diff
@@ -3557,17 +3557,18 @@ package body Templates_Parser is
                -- Rewrite --
                -------------
 
-               procedure Rewrite (T : in out Tree) is
+               procedure Rewrite (T : Tree) is
+                  L : Tree := T;
```
Comment (leogermond) [inline src/templates_parser.adb:3561]: I don't understand the use of a temporary `Tree` object?

**Review by leogermond — CHANGES_REQUESTED** [review]

File: src/templates_parser-print_tree.adb (hunk 1 above)
Comment (TurboGit) [reply src/templates_parser-print_tree.adb]: Done.

File: src/templates_parser.adb (hunk 3 above)
Comment (TurboGit) [reply src/templates_parser.adb:3561]: I don't want to change T (was in out) as we do change this pointer (see Move_To_Last). And this was so changing the pointer of if true/false sub-tree:

```
                           Rewrite (L.N_True);
                           Rewrite (L.N_False);
```

So changing the actual tree, which was fully wrong.

**Review by leogermond — APPROVED** [review]

**Pull request merged**